calicoctl/calicoctl/commands/ipam/split.go

```diff
@@ -0,0 +1,208 @@
+	}
+
+	// Disable the specified IP pool.
+	oldPool.Spec.Disabled = true
+	oldPool, err = client.IPPools().Update(ctx, oldPool, options.SetOptions{})
+	if err != nil {
+		return fmt.Errorf("Error disabling IP pool %s", parsedArgs["<CIDR>"])
+	}
+
+	// Calculate the split pool CIDRs.
+	numString := parsedArgs["<NUMBER>"].(string)
+	splitNum, err := strconv.Atoi(numString)
+	if err != nil {
+		return fmt.Errorf("Error reading number to split IP pools into. %s is not a valid number: %v", numString, err)
+	}
+
+	splitCIDRs, err := splitCIDR(oldPool.Spec.CIDR, splitNum)
+	if err != nil {
+		return fmt.Errorf("Error splitting the CIDR %s into %d CIDRs: %v", oldPool.Spec.CIDR, splitNum, err)
+	}
+
+	// Create the new split pools using UnsafeCreate.
+	for i, cidr := range splitCIDRs {
+		splitPool := &apiv3.IPPool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("split-%s-%d", oldPool.GetObjectMeta().GetName(), i+1),
+			},
+			Spec: apiv3.IPPoolSpec{
+				CIDR: cidr,
+				// All other fields are preserved.
+				BlockSize:        oldPool.Spec.BlockSize,
+				VXLANMode:        oldPool.Spec.VXLANMode,
+				IPIPMode:         oldPool.Spec.IPIPMode,
+				NATOutgoing:      oldPool.Spec.NATOutgoing,
+				DisableBGPExport: oldPool.Spec.DisableBGPExport,
+				NodeSelector:     oldPool.Spec.NodeSelector,
+				AllowedUses:      oldPool.Spec.AllowedUses,
+			},
+		}
+
+		_, err = client.IPPools().UnsafeCreate(ctx, splitPool, options.SetOptions{})
+		if err != nil {
+			return fmt.Errorf("Error using unsafe create to make split pool with cidr %s: %v", cidr, err)
+		}
+	}
+
+	// Delete the old IP pool.
+	_, err = client.IPPools().Delete(ctx, oldPool.GetObjectMeta().GetName(), options.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("Error removing the IP pool that was split %s: %v", oldPool.GetObjectMeta().GetName(), err)
+	}
+
+	// Output follow-up directions.
+	fmt.Printf("IP Pool %s was successfully split into %d smaller pools. Please refer to the documentation for final steps.", oldPool.GetObjectMeta().GetName(), splitNum)
+
+	return nil
+}
+
+func splitCIDR(oldCIDR string, parts int) ([]string, error) {
+	// Validate that we are trying to split the CIDR into a valid number of child CIDRs.
+	power := math.Log2(float64(parts))
+	if math.IsNaN(power) || power == 0 {
+		return []string{}, fmt.Errorf("Number to split CIDR into is not a valid power of 2: %d", parts)
+	}
+
+	// Convert the string version of the CIDR into a CIDR object.
+	_, cidr, err := cnet.ParseCIDR(oldCIDR)
+	if err != nil {
+		return []string{}, fmt.Errorf("Error reading CIDR %s before attempting to split it", oldCIDR)
+	}
+
+	ones, bits := cidr.Mask.Size()
+	if int(power)+ones > bits {
+		return []string{}, fmt.Errorf("The CIDR %s is not large enough to be split into %d parts", oldCIDR, parts)
+	}
+
+	// Find the block size to increment over.
+	blockSize := new(big.Int).Exp(big.NewInt(2), big.NewInt(int64(bits-ones-int(power))), nil)
```

Comment:
`blockSize` is perhaps a confusing name for this, given it already has a meaning in the IPAM space. Maybe `newPoolSize` would be better?